src/analysis/modifications/moment.jl

```diff
@@ -0,0 +1,110 @@
+"""
+add_moment_constraints(
+    kcats::Dict{String,Float64},
+    molar_weights::Dict{String,Float64},
+    protein_mass_fraction::Float64;
+    protected_constraint_ids = ["ATPM"],
+)
+
+Add enzyme capacity constraints to problem using the MOMENT algorithm. Requires `kcats` and
+`molar_weights` for each reaction and protein. Proteins are encoded by their associated gene.
+Adds a variable vector `y` to the problem corresponding to the protein concentration. Adds a
+variable vector `t` that is internal to the problem - ignore. Set the `protein_mass_fraction`
+capacity constraint to the fraction of the cell mass that is composed of proteins.
+Optionally `protected_constraint_ids` can be used to ensure that some reaction fluxes are not
```

Comment:
"some are not unconstrained" too much negation, perhaps

Optionally it is possible to prevent constraining of some of the reaction fluxes by specifying the reaction IDs in `protected_constraint_ids`. ?

shouldn't it be `protected_rids` ?